core/database/foxx/tests/record.test.js

```diff
@@ -0,0 +1,341 @@
+"use strict";
+
+const chai = require("chai");
+const expect = chai.expect;
+const Record = require("../api/record");
+const g_db = require("@arangodb").db;
+const g_lib = require("../api/support");
+const arangodb = require("@arangodb");
+
+describe("Record Class", () => {
+  beforeEach(() => {
+    g_db.d.truncate();
+    g_db.alloc.truncate();
+    g_db.loc.truncate();
+    g_db.repo.truncate();
+  });
+
+  it("unit_record: should initialize correctly and check record existence is invalid", () => {
+    const record = new Record("invalidKey");
+    expect(record.exists()).to.be.false;
+    expect(record.key()).to.equal("invalidKey");
+    expect(record.error()).to.equal(g_lib.ERR_NOT_FOUND);
+    expect(record.errorMessage()).to.equal(
+      "Invalid key: (invalidKey). No record found.",
+    );
+  });
+
+  it("unit_record: should initialize correctly and check record existence is valid", () => {
+    const valid_key = "1111";
+    const key_id = "d/1111";
+    const owner_id = "u/bob";
+    const repo_id = "repo/datafed-at-com";
+    // Create nodes
+    g_db.d.save({
+      _key: valid_key,
+      _id: key_id,
+    });
+    g_db.repo.save({
+      _id: repo_id,
+    });
+    g_db.u.save({
+      _id: owner_id,
+    });
+
+    // Create edges
+    g_db.loc.save({
+      _from: key_id,
+      _to: repo_id,
+      uid: owner_id,
+    });
+    g_db.alloc.save({
+      _from: owner_id,
+      _to: repo_id,
+    });
```

Comment:
If this setup is common we can pull it out as a function and pass in what we need ahead.